compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs

```diff
@@ -94,7 +94,13 @@ pub fn push_debuginfo_type_name<'tcx>(
             push_debuginfo_type_name(tcx, inner_type, true, output, visited);
 
             if cpp_like_names {
-                output.push('*');
+                // Slice and Str refs are treated as equivalence of C++ pointer when computing debug
+                // info for MSVC debugger. However, adding '*' at the end of these types' type names
+                // will cause the .natvis engine for WinDbg fails to display data of Slice and Str objects
```

Comment:
```suggestion
                // Slices and `&str` are treated like C++ pointers when computing debug
                // info for MSVC debugger. However, adding '*' at the end of these types' names
                // causes the .natvis engine for WinDbg to fail to display their data, so we opt these
                // types out to aid debugging in MSVC.
```
Slightly altered the comment to make it a little clearer. If you're happy with this, could you squash this change into your comment, and then it looks good, thanks!